### PR TITLE
Use Spark archive site to download for a broader range of versions that can be used.

### PR DESCRIPTION
**Pull request by creechy**
The main download site does not include "archived" versions.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -26,7 +26,7 @@ RUN mkdir -p ${HADOOP_HOME} && mkdir -p ${SPARK_HOME}
 WORKDIR ${SPARK_HOME}
 
 # Download and install Spark
-RUN curl https://dlcdn.apache.org/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop3.tgz -o spark-${SPARK_VERSION}-bin-hadoop3.tgz \
+RUN curl https://archive.apache.org/dist/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop3.tgz -o spark-${SPARK_VERSION}-bin-hadoop3.tgz \
```
Comment (defalut) [inline Dockerfile]: This worked for me
```suggestion
RUN curl https://archive.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop3.tgz -o spark-${SPARK_VERSION}-bin-hadoop3.tgz \
```

Comment (creechy) [reply Dockerfile]: Yeah my bad. I tried to create a PR on the fly on the web. You have the correct URL.

**Comment by mrn-aglic** [comment]
@creechy  Sorry it took so long for me to take a look.
Does the archive site contain the newest versions as well?

**Comment by creechy** [comment]
> @creechy Sorry it took so long for me to take a look. Does the archive site contain the newest versions as well?

@mrn-aglic yes I believe it does. it might be a little slower than the main site but I believe all versions are there.

**Pull request merged**